library/manageiq_provider.py

```diff
@@ -48,19 +48,19 @@
       - the provider zone name in manageiq
     required: false
     default: null
-  hostname:
+  provider_api_hostname:
     description:
-      - the added provider hostname
+      - the added provider hostname API
```

Comment:
Perhaps "the added provider API hostname" or "hostname of the added provider API" ?

Also since the module can modify existing providers, I'd replace s/added provider/provider/ everywhere.